charts/ovn/templates/daemonset-controller.yaml

```diff
@@ -55,19 +81,36 @@ spec:
     spec:
       serviceAccountName: {{ $serviceAccountName }}
       hostNetwork: true
-      dnsPolicy: {{ .Values.pod.dns_policy }}
+      hostPID: true
+      hostIPC: true
+      dnsPolicy: ClusterFirstWithHostNet
       nodeSelector:
         {{ .Values.labels.ovn_controller.node_selector_key }}: {{ .Values.labels.ovn_controller.node_selector_value }}
       initContainers:
 {{- tuple $envAll "ovn_controller" list | include "helm-toolkit.snippets.kubernetes_entrypoint_init_container" | indent 8 }}
+        - name: get-gw-enabled
+{{ tuple $envAll "ovn_controller_kubectl" | include "helm-toolkit.snippets.image" | indent 10 }}
+          command:
+            - /bin/bash
+            - -c
+            - |
+              kubectl get node ${NODENAME} -o jsonpath='{.metadata.labels.l3-agent}' > /tmp/gw-enabled/gw-enabled
+          env:
+            - name: NODENAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: gw-enabled
+              mountPath: /tmp/gw-enabled
+              readOnly: false
```

Comment:
This breaks our current process, since I don't think we use `l3-agent` label to decide what's gateway and what's not..